crates/matrix-sdk/src/widget_api/api/widget.rs

```diff
@@ -0,0 +1,77 @@
+use async_trait::async_trait;
+use tokio::sync::{
+    mpsc::{UnboundedReceiver as Receiver, UnboundedSender as Sender},
+    oneshot,
+};
+use uuid::Uuid;
+
+use super::{
+    super::{
+        handler::{OutgoingMessage, Widget as WidgetSender},
+        messages::{to_widget::ToWidgetMessage, Header},
+        Error,
+    },
+    PendingResponses, Result,
+};
+
+#[derive(Debug)]
+pub struct Widget {
+    pub info: Info,
+    pub comm: Comm,
+}
```

Comment:
Sorry, I did not quite get what you mean 🙂 

The `api::widget::Widget` describes an input parameter to the `api` module, so that it knows how to communicate with a widget (I did not touch this part in this PR, it's just part of the merge/rebase). This is something that a caller/user of the `api` module does in order to start the Client Widget API state machine.

The `WidgetSink` OTOH is the structure internal to the `api` module that takes over the logic of receiving outgoing messages, converting them to the `ToWidget` messages, and waiting for responses. Perhaps the right call would be to move `WidgetSink` out of the `widget` module and make it non-`pub` and rename it to something less ambiguous.

Reply:
Refactored.